instant-xml-macros/src/de.rs

```diff
@@ -1,11 +1,54 @@
 use proc_macro2::{Ident, Span, TokenStream};
 use quote::{quote, ToTokens};
 
-use super::{discard_lifetimes, ContainerMeta, FieldMeta, Namespace};
+use super::{discard_lifetimes, ContainerMeta, FieldMeta, Namespace, VariantMeta};
 
 pub(crate) fn from_xml(input: &syn::DeriveInput) -> TokenStream {
     let ident = &input.ident;
     let container_meta = ContainerMeta::from_derive(input);
+    match &input.data {
+        syn::Data::Struct(ref data) => deserialize_struct(input, data, container_meta, ident),
```

Comment:
For now, I'd like `Struct` variants to yield an error if `container_meta.scalar` is `true`.

Also, please separate out the change to extract a `deserialize_struct()` function.